### PR TITLE
Support TypeScript 3.7 (and drop Node 6 support for direct inst…

### LINKED ISSUE #6569: [Typescript] Prettier removes question mark from optional computed class property
**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAjAhgJ0wF5MByPfUgbgB0owAbXNNTAMQgk2Dsz8wDaFALoB+WlH6CAjGIlSB5XAC9ScugF86IADQgIABxgBLdMlAF8EAO4AFAgjTIQuAG4RjAE10hs+XGAA1nAwAMoGAcZQAObIMPgArnB6ABYwALYMAOopxvBoEWBwoY55xq55AJ7OYCw+UWhw+DC2-tHpuMgAZrgMjXoAVmgAHgBC-kEhobjpcAAyUXDdvf0gQ8OhUdEMcACKCRDwy33JIBH4jfjOMJUGcGhg+MZGPgZPsFleMCnIABwADHo3hBGll-AZnG97k1XEs9ABHA7wVqGJwoZgAWigcDgnlxPnwcERxkJrVw7U6SB6Jz0jXSxjiiVOaC2O32hyWVJWpxguGwn0832QACY9PFcMYGFsAMIQdIdZxQaBwkAJRoAFT5aOpjQ0GiAA)
```sh
--parser typescript
```

**Input:**
```tsx
const bar = 'bar';
class Foo {
    [bar]?;
    [1]?;
    ['baz']?;
}

```

**Output:**
```tsx
const bar = "bar";
class Foo {
  [bar];
  [1]?;
  ["baz"]?;
}

```

**Expected behavior:**
Prettier should preserve ? in output. Expected output:
```tsx
const bar = "bar";

class Foo {
  [bar]?;
  [1]?;
  ["baz"]?;
}

```

This only happens when the computed property is a variable.

### LINKED ISSUE #6574: [TypeScript 3.7] add support for `asserts` keyword
**Prettier 1.18.2**

TypeScript is developing support for [assertions in control flow analysis](https://github.com/microsoft/TypeScript/pull/32695). The functionality is now available in development builds such as [v3.7.0-dev.20190928](https://www.npmjs.com/package/typescript/v/3.7.0-dev.20190928). This ticket requests enabling Prettier support of the new `asserts` keyword.

[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAhgM7FwBOMAFAG6EA26cS+mA1lBAO5QCULJMpWL46jOPmAAdKPnwwAFuW74ocLvgCi5ZeSq8ZAXxkgANCAgAHXNGLJQhHdwAKjhHZSEaEHABMzIABG5IRgbHAwAMqWoThQAObIMORM5gowALb0AOoKOPDEMWBwke75ODT5AJ7I4KQBcUIwziHxGYTIqAxk5gBWxAAeAEIhYRGRhBlwADJxcJ3dcH2DkXHx9HAAiugQ8Av0PSAx5EK1MFWWcMRg5DjWAZa3sNl+isgAHAAM5o8QZNkhSy1R5XCg0ebmACOO3gLSsHhAJAAtGo4L40QFyHBoTgsS1CG0OkgugcliAyBkcEkUmTiGsNttdvNiYtzDBCIEXr43kgAExskI4ehrADCEAy7VqnDUAXQZAAKhyPCSyIZDEA)
```sh
--parser typescript
```

**Input:**
```tsx
function assert(value: unknown): asserts value {
  throw new Error()
}
```

**Output:**
```tsx
SyntaxError: '{' or ';' expected. (1:42)
> 1 | function assert(value: unknown): asserts value {
    |                                          ^
  2 |   throw new Error()
  3 | }
  4 | 
```

**Expected behavior:**
```tsx
function assert(value: unknown): asserts value {
  throw new Error()
}
```

### LINKED ISSUE #6595: [TypeScript 3.7] add support for optional chaining
**Environments:**
- Prettier Version: 1.18.2
- Running Prettier via: CLI

**Steps to reproduce:**
I was able to reproduce in the playground when using the typescript parser, which I guess it's the parser that ends up being used by prettier in typescript files? In my case its happening for both `.ts` and `.tsx` file extensions.

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDAdAIwPzrCAGhAgAcYBLaAZ2VFQCd6IB3ABQYRpVQDcJyAJoRCZ6qMAGs4MAMolx5KAHNkMegFc4RABYwAtgBsA6tvLwq8sHBmcz5HmYCeycFRpFFVOPRisxSvVRkADNUAy8iACsqAA8AITFJaRlUPTgAGUU4ELCIkGiYmUUlAzgARXUIeBzwrRB5ei96FxhHEjgqMHpyMmESbtgjQRhtZAAOAAYifogvIzESF36O7x5sogBHSvg-Ui40KgBaKDg4ATPhejgt8iu-VACgpFDaoi89clUNOqpi0oqqtlnrk6jBUJghgIRsgAExENSocgGYoAYQgekCLig0HWIHUXgAKuCuC8vABfMlAA

**Expected behavior:** does not throw "Expression expected" syntax error when using optional chaining in typescript files

**Actual behavior:** throw "Expression expected" syntax error when using optional chaining in typescript files

### LINKED ISSUE #6609: [TypeScript 3.7] add support for nullish coalescing operator
**Prettier 1.18.2**

**Input:**
```jsx
const a = null
console.log(a ?? "keque")
```

**Output:**
```
[error] src/index.tsx: SyntaxError: Expression expected. (8:16)
[error]    6 | 
[error]    7 | const a = null
[error] >  8 | console.log(a ?? "keque")
[error]      |                ^
[error]    9 | 
```

**Expected behavior:**
Should understand or at least ignore such expression.

### LINKED ISSUE #6661: [TypeScript 3.7] declare modifier on class fields
see https://github.com/microsoft/TypeScript/pull/33509

**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgMI5wAe8UAJrgEI7AA6UOOZcYGATnDgA5I4BMAVnoBfeiAA0ICFxgBLaJmSh0bNhADuABRUJFKdADcIsshJAAjNujABrODADKXa7KgBzZDDYBXOJIAWMAC2qADq-rLwmM5gcA66kbIGkQCeyODYZq6YcGwwmlZuQejIAGboqDmSAFaYRJRWtvYO6EFwADKucGUVVSC1RA6ubqhwAIreEPA9lX4gzmw5bOkwKVxwmGBssjJmXNuwoSYw-sgAHAAMkvsQOaFWXOn7G7kG3ZIAjpPwBdJ6IFgALRQOBwZimSQcL6yDgFdBFEpIcqzSQ5IKyTw+OaYYajCZTbpI3pzGDocxHMgnZB8SRedCyVDDPAQILFdJQaDvEDeHIAFTJemROWEwiAA)
```sh
--parser typescript
```

**Input:**
```tsx
class C extends B {
  declare p: 25
}

```

**Output:**
```tsx
class C extends B {
  p: 25;
}

```

**Expected behavior:**

same as the input

### PR DESCRIPTION
**Pull request by Cryrivers**
This PR addresses TypeScript 3.7 supports, which includes:
- Optional Chaining (Fixes #6595)
- Null Coalescing (Fixes #6609)
- `asserts` keyword (Fixes #6574)
- `declare` keyword on class fields (Fixes #6661)

Fixes #6569

- [X] I’ve added tests to confirm my change works.
- [X] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [X] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [X] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://deploy-preview-6657--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEMATgJ4DicMAYjkQM4wCiANnALYIwA8AKgHwAKAIZEiAfiT4eAbQC6ASnzAAOlHzr8RSuiJqR4gHTSADLIDcqgL6qNtu-Yf3VQuiSz4MWXATZCA1nAASnAAjuhwDAI6zJIMRDhQAOYANPjMEIkS+AJsdImxxAmJigC8fPgAbhA4ACaKKmoa6ZkGAgAGwWERMPgMIvA1+EI9ACTAUHAA7vgAIsNwAvIGMBAAkgDKAPLrhUmLlm3yFo3qkFAMmhHozD0l2UKTQjg9qJRgABZRRMzySwBWdGgi2OtmaYlaHVC4QuqASODo7zgg2GQ1G4yms3mi2Way2O3ie3kByONg0WhgOjUWjo1xgx2sUFUrB6AA98HdUBAIPgxGJ8AAjETA1SqTzYPB6Oh0OBEGCrOgzOCwiY1XiCCpCGJSeSSFzS2V0Sqa-Dw-AAOWgZuuzCE-NYaqUpPUOFQ2Q1zHZJTumBqSoSSPwAB9A0aPV67lBrfUnXYYO8iBBphNpgBBKUy7xQRiiCBEAQxxxtRgsgAOcGwAYA5O7K4Rufy4Phfcqkal+egelowHAcBUA2N3QcCxoSSd8AyGSBkiAICXM3RkKB9ImAAoiBALlBCKq1KcgflEIRgAIwdYlo9FZDEcLT94wNjMADq72eEXP3fWG+eveeJGQ4ClPcEn1GAV0PRJfGQVBNWlacARZAAhQ9j0odYhA4AAZf0oJguA4LoFl1iKVgAEV0AgeAcOYWCQHPegZX-GASDLOgwHiOc9xLAkYEfWo42QAAOYxpy4iBpUfQ8S3-LiIhlPs9zCCi4DA2dNxAFwAFoJiRJE9y0MJaGU8DIKQaDqLwkBpTYHAryIG9LOIuAyKUqiaJgW1eJqfikAAJmnYgnmYIoAGEIDYEyQCgaA4D3dBpR4W1NzMmiKnCVYoF9WB1jYnA5xTDKdhIVhXLgSxLCAA)✨**

### CODE REVIEW

**Comment by lydell** [comment]
Looks like your snapshots aren’t up-to-date?

**Comment by Cryrivers** [comment]
@lydell Yeah, it's still WIP.

**Comment by Cryrivers** [comment]
@suchipi fyi

**Comment by Cryrivers** [comment]
@lydell updated

**Comment by Cryrivers** [comment]
btw `@typescript-eslint/estree` has `chokidar` as a dependency for watch feature, which doesn't make `rollup` happy. could i get any advice for that?

**Comment by lydell** [comment]
I think it should be possible to stub chokidar out. Maybe @fisker can help?

**Comment by fisker** [comment]
@lydell Even if we can make this work, still new version `@typescript-eslint/typescript-estree` can't update due to node version problem

like I said in #6601

> Notice: @typescript-eslint/typescript-estree@2.3.2 requires node >=8 , so this PR might can't merge until prettier@2

**Comment by lydell** [comment]
Ah, thanks, I forgot about that.

**Comment by radiosilence** [comment]
> @lydell Even if we can make this work, still new version `@typescript-eslint/typescript-estree` can't update due to node version problem
> 
> like I said in #6601
> 
> > Notice: @typescript-eslint/typescript-estree@2.3.2 requires node >=8 , so this PR might can't merge until prettier@2

How long might that be? Surely node 6 counts as legacy now. Node 8 was released 2 years ago.

**Comment by SimenB** [comment]
>I'm not sure node 6 is even in its LTS maintenance window any more.

6 has been EOL since April. (12 will be LTS Monday next week)

**Comment by Cryrivers** [comment]
@duailibe @ikatyang could i know what you think about dropping the support for Node 6?

**Comment by fisker** [comment]
https://github.com/prettier/prettier/issues/3919

@Cryrivers

**Comment by radiosilence** [comment]
> > I'm not sure node 6 is even in its LTS maintenance window any more.
> 
> 6 has been EOL since April. (12 will be LTS Monday next week)
> 
> https://github.com/nodejs/Release/blob/master/README.md

Can we drop support for it then?

**Comment by lydell** [comment]
> drop node 6 for direct-from-GitHub installs

Sounds good to me!

**Comment by Cryrivers** [comment]
@fisker could you help advice how to shim `fs-events`?

**Comment by fisker** [comment]
@Cryrivers 

I already merged your code into #6680 about how to make it work, you can ready my PR there is build script in another repo.

**Comment by Cryrivers** [comment]
@fisker great, i'll close my PR when yours has been merged.

**Comment by Cryrivers** [comment]
FYI, `typescript-estree` removed the dependency on `chokidar` in `2.5.0`

**Comment by Cryrivers** [comment]
Now you can try in the playground for this PR!

**Comment by fisker** [comment]
actually Its your call,  drop node 6, merge this.

support node 6, merge #6680. 

they are the same

**Comment by evilebottnawi** [comment]
@fisker 
> actually Its your call, drop node 6, merge this.

It is just proposal, we need consider this

/cc @prettier/core Which solution will we choose?

**Comment by Cryrivers** [comment]
@evilebottnawi as long as we can deliver before TypeScript 3.7 lands on, i'm fine with either PR, too. imho, i prefer dropping the support of Node.js 6 so we dont have to use customized build of `typescript-estree`.

**Comment by thorn0** [comment]
@evilebottnawi see also https://github.com/typescript-eslint/typescript-eslint/issues/1108

**Comment by bradzacher** [comment]
also note, we haven't yet added support for the ts declare keyword on class members - https://github.com/typescript-eslint/typescript-eslint/issues/1086

**Comment by Cryrivers** [comment]
I’ll go through the new features in 3.7 and add accordingly.

**Comment by Cryrivers** [comment]
> also note, we haven't yet added support for the 3.7's declare keyword on class members - [typescript-eslint/typescript-eslint#1086](https://github.com/typescript-eslint/typescript-eslint/issues/1086)

sorry i didn't find it in the TypeScript 3.7 Beta announcement post. Could you help by posting the PR link in TypeScript repo?

**Comment by thorn0** [comment]
@Cryrivers Announcement posts ≠ changelog. See https://github.com/microsoft/TypeScript/pull/33509

**Comment by Cryrivers** [comment]
yep. i'll also check if private class fields is in 3.7 as well

**Comment by Cryrivers** [comment]
> also note, we haven't yet added support for the 3.7's declare keyword on class members - [typescript-eslint/typescript-eslint#1086](https://github.com/typescript-eslint/typescript-eslint/issues/1086)

i just looked into it and found it quite tricky. reason being:

1. `typescript-estree` reuses `TSDeclareFunction` type for the `declare` keyword in class properties. I'm not sure if it is a bug.

2. Prettier has a rule that it only prints out `declare` if the `TSDeclareFunction` node has a truthy `declare` property. Here's the comment in code:
> // For TypeScript the TSDeclareFunction node shares the AST
> // structure with FunctionDeclaration

3. `TSDeclareFunction` node for class properties has no truthy `declare` property.

**Comment by Cryrivers** [comment]
@bradzacher i just looked into it. Right now we are unable to visit the AST node for that `declare` keyword, or have `declare` flag in `ClassProperty` node. So we probably might want to wait for `typescript-estree` to address it first.

**Comment by evilebottnawi** [comment]
/cc @Cryrivers What is status?

/cc @prettier/core We should solve this problem asap, our release has already lagged behind the planned date very much

We have two PRs https://github.com/prettier/prettier/pull/6680 and this

What is our final decision?
- Drop node 6 for direct-from-GitHub installs (prefer)
- Using fork?

**Comment by Cryrivers** [comment]
this branch is ready for most TypeScript users (except for the support of declared class property, which is not yet supported by typescript-eslint/typescript-eslint#1086)

**Comment by nodkz** [comment]
Merge! Merge! Merge! 🔥

@Cryrivers Incredible work 👍
I hope that you ship new package ASAP and nothing will not stop you. 😉

**Comment by evilebottnawi** [comment]
Will be great fix https://github.com/typescript-eslint/typescript-eslint/issues/1086 before merge to implement full support 3.7

**Comment by sosukesuzuki** [comment]
I think it is better to drop node 6 for direct-from-GitHub installs than using fork.

**Comment by evilebottnawi** [comment]
https://github.com/prettier/prettier/pull/6657#issuecomment-545873993 -
:+1: for drop
:rocket: for fork

**Comment by Cryrivers** [comment]
> As workaround we can use regexp 💡

imho i prefer waiting for `typescript-estree` addressing the issue first so no hacky workarounds. i feel i can indicate it in `CHANGELOG.md`.

File: CHANGELOG.unreleased.md [hunk 1]
```diff
@@ -44,6 +44,16 @@ const link = <a href="example.com">http://example.com</a>;
...
+#### TypeScript: Support TypeScript 3.7 syntax features ([#6657] by [@Cryrivers])
+
+Add following syntax support for TypeScript 3.7:
+
+- Optional Chaining
+- Null Coalescing
+- `asserts` keyword
+
+NOTE: `declare` keyword on class members has not been supported yet, which will be fixed in future versions.
+
```
Comment (evilebottnawi) [inline CHANGELOG.unreleased.md]: Let's add examples for each feature as we do in CHANGELOG for other issues - Core/`Output (Prettier stable)`/`Output (Prettier master)`

**Comment by nodkz** [comment]
Ooop, just retweeted on 11k subscribers
https://twitter.com/nodkz/status/1187373836112711682?s=19

File: CHANGELOG.unreleased.md (hunk 1 above)
Comment (Cryrivers) [reply CHANGELOG.unreleased.md]: > Let's add examples for each feature as we do in CHANGELOG for other issues - Core/`Output (Prettier stable)`/`Output (Prettier master)`

Done :)

**Comment by JamesHenry** [comment]
FYI I merged the declare keyword support, new version will be cut automatically in a few minutes

**Comment by JamesHenry** [comment]
(Thanks to @bradzacher for actually doing the work)

**Comment by Cryrivers** [comment]
> FYI I merged the declare keyword support, new version will be cut automatically in a few minutes

supporting code added, waiting for the release :)

**Comment by thorn0** [comment]
[Playground is broken](https://deploy-preview-6657--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgB47AA6UOZJAviADQgQAOMAltJsqOgE6cQDuAClwRsU6AG4QmAExogARp3RgA1nBgBlekqZQA5shicArnFoALGAFtUAdTNN4mLWDjrhDpmIcBPZOGyyOphwnDD8irqW6MgAZuiowbQAVph4AEKKKmrq6JZwADI6cLHxiSApeOo6uqhwAIpGEPAlCaYgWpzBnH4w3vRwmGCcTIyy9MOwNtIwZsgAHAAMtOMQwTaK9H7jAyFixbQAjo3w4QwiIFgAtFBwcFJ3spxwR0xP4eiR0UhxrbTBlkwDMY2phqrUGk1it9Sm0YOg5FMpDNkAAmWiGdBMVDVADCEEsUT8UGg+xARmCABV4SIfmUxCYAJJQe6wdRDEYwACCzPUvVqLWCFAoQA).

File: CHANGELOG.unreleased.md [hunk 2]
```diff
@@ -44,6 +44,115 @@ const link = <a href="example.com">http://example.com</a>;
...
 -->
 
+#### TypeScript: Support TypeScript 3.7 syntax features ([#6657] by [@cryrivers])
+
+Add following syntax support for TypeScript 3.7:
+
+- Optional Chaining
+- Null Coalescing
+- `asserts` Keyword
+- `declare` keyword on class members
```
Comment (thorn0) [inline CHANGELOG.unreleased.md]: "on class fields", not members. Please change everywhere.

File: CHANGELOG.unreleased.md [hunk 3]
```diff
@@ -44,6 +44,115 @@ const link = <a href="example.com">http://example.com</a>;
...
+
+##### Optional Chaining
+
+Previous versions would not be able to format this syntax, this has been fixed in this version.
```
Comment (thorn0) [inline CHANGELOG.unreleased.md]: This description is not needed. The title suffices for the description. (Please see the comments in the beginning of `CHANGELOG.unreleased.md`.)

File: package.json [hunk 4]
```diff
@@ -67,7 +67,7 @@
     "resolve": "1.12.0",
     "semver": "6.3.0",
     "string-width": "3.1.0",
-    "typescript": "3.6.4",
+    "typescript": "3.8.0-dev.20191024",
```
Comment (thorn0) [inline package.json]: Please switch to `3.7.1-rc`. `3.8.0-dev` can (and seemingly does) contain new breaking changes.

**Comment by thorn0** [comment]
Also it'd be good to have a separate changelog entry about dropping Node 6.

File: package.json (hunk 4 above)
Comment (Cryrivers) [reply package.json]: oh, accidentally upgraded to 3.8.0

**Comment by Cryrivers** [comment]
> See my comments. Also it'd be good to have a separate changelog entry about dropping Node 6.

@thorn0 most of comments should be addressed.

**Comment by thorn0** [comment]
@Cryrivers  why do the tests fail?

**Comment by Cryrivers** [comment]
@JamesHenry may i know the changes of `typescript-estree@2.5.1-alpha.3`? i'm pretty sure it breaks the prettier dist now.

**Comment by bradzacher** [comment]
`typescript-estree@2.5.1-alpha.3` is the release containing the declare class support.
There have been no other changes since that was merged and auto-released 13 hours ago.

https://github.com/typescript-eslint/typescript-eslint/commits/master

**Comment by Cryrivers** [comment]
@bradzacher thanks.

**Comment by thorn0** [comment]
@bradzacher (This)[https://github.com/typescript-eslint/typescript-eslint/pull/1111/files#diff-ecbfe9d2916e703f957488c869327e21R23] doesn't work in the browser because we don't have `ts.sys` there.

**Comment by thorn0** [comment]
Oops! In `(a?.b).c` both dots are parsed as `OptionalMemberExpression`.
@bradzacher

**Comment by thorn0** [comment]
This PR is blocked by https://github.com/typescript-eslint/typescript-eslint/issues/1139

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/pull/1141 was just merged, which should fix the AST, and better support browser environments.
(will be released to canary tag in ~15 mins).

**Comment by fisker** [comment]
Do we need change this?

https://github.com/prettier/prettier/blob/4e46f92b8648c7f25de62bfaf4368aa3f2e588d9/package.json#L14

**Comment by thorn0** [comment]
@sosukesuzuki Optional chaining is going to become much more popular after the TS 3.7 gets released, so better to catch all these parens issues in this PR and be ready.

**Comment by ikatyang** [comment]
@Cryrivers 

> @ikatyang could i know what you think about dropping the support for Node 6?

Sorry for the late response. FWIW, I'm in favor of dropping legacy Node, but we've promised to our user that we won't make breaking changes in minor/patch release (according to semver) so we shouldn't introduce such change in minor release IMO, though I'm still not quite sure if this rule should be applied here. One possible backward-compatible workaround would be to introduce some `postinstall` scripts to hide those breaking changes until 2.0.

**Comment by lydell** [comment]
@ikatyang Just in case you weren’t aware, we only drop Node 6 support **for direct installation from GitHub.** The version released on npm will still work – even in Node.js 4. See also: https://github.com/typescript-eslint/typescript-eslint/issues/1108#issuecomment-545275953

Not sure if that changes your opinion, but just in case 🤷‍♂

**Comment by ikatyang** [comment]
Yeah, I'm aware. I meant that we've explicitly specified `node>=6` in `package.json`, which looks like a promise to me, I'm not sure how many people use it this way though.

**Comment by thorn0** [comment]
@ikatyang IMHO the contents of package.json _on GitHub_ can't be a promise unless such a promise is documented somehow, otherwise we could call any piece of code a promise.

**Comment by ikatyang** [comment]
@thorn0 Yes, but we did _mention_ that we support installation from GitHub in https://prettier.io/versions.html, that's why I'm still not sure if it is considered a promise. Anyway, this is just a suggestion, not a requirement.

**Comment by thorn0** [comment]
@ikatyang It's called "next" there though, not 1.x.x.

**Comment by soroushm** [comment]
So we are waiting to release :)

**Comment by kirananto** [comment]
Awesome.. waiting for the release.. 💯

**Comment by evilebottnawi** [comment]
@ikatyang I agree with @thorn0 , next also can be unstable, for example for `2.0` we also will be have `next`, but with breaking changes

**Comment by deser** [comment]
Guys, when it's planned to deliver this to npm?

**Comment by damiangreen** [comment]
Is anyone getting `Error: Cannot find module 'source-map-support'` issues with this?

**Comment by thorn0** [comment]
@damiangreen What exactly are you doing to get it?

**Comment by damiangreen** [comment]
@thorn0 It appears to happen on our Bamboo build server.

**Comment by thorn0** [comment]
@damiangreen What exactly is that build server doing when this happens? Do you have any stack traces or something?

**Comment by damiangreen** [comment]
@thorn0 Our devops jsut pulled tis down from AWS Console. 
I jsut switched back from `    "prettier": "https://github.com/Cryrivers/prettier.git", `to using `  "prettier": "^1.18.2",` and the error went away.

Note I have the latest  "source-map-support": "^0.5.15", in my `dependencies

```
09:01:51
internal/modules/cjs/loader.js:638

09:01:51
throw err;

09:01:51
^

09:01:51
Error: Cannot find module 'source-map-support'

09:01:51
at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)

09:01:51
at Function.Module._load (internal/modules/cjs/loader.js:562:25)

09:01:51
at Module.require (internal/modules/cjs/loader.js:692:17)

09:01:51
at require (internal/modules/cjs/helpers.js:25:18)

09:01:51
at Object.<anonymous> (/usr/src/app/server.js:1:1)

09:01:51
at Module._compile (internal/modules/cjs/loader.js:778:30)

09:01:51
at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)

09:01:51
at Module.load (internal/modules/cjs/loader.js:653:32)

09:01:51
at tryModuleLoad (internal/modules/cjs/loader.js:593:12)

09:01:51
at Function.Module._load (internal/modules/cjs/loader.js:585:3)
```

**Comment by thorn0** [comment]
@damiangreen You still haven't written what this process is doing. How does it use Prettier? Is this stack trace full?

**Pull request merged**